### PR TITLE
Featureform in a single container

### PR DESCRIPTION
**Pull request by sdreyer**
# Description

This pull request adds featureform into a single container using nginx and supervisord. 
The goal is to decrease the time to run featureform locally

### Does this correspond to an open issue?

No

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

**Comment by sdreyer** [comment]
@Sami1309 if you have time to look over this again

File: newserving/Dockerfile [hunk 1]
```diff
@@ -29,5 +29,6 @@ FROM alpine
 COPY --from=0 ./app/main ./main
 COPY --from=0 ./app/provider/scripts/offline_store_spark_runner.py /app/provider/scripts/offline_store_spark_runner.py
 
+ENV SERVING_PORT "0.0.0.0:8080"
```
Comment (Sami1309) [inline newserving/Dockerfile:32]: would this make address 0.0.0.0:0.0.0.0:8080 then?

File: metadata/server/server.go [hunk 2]
```diff
@@ -6,38 +6,49 @@ package main
...
-		Logger:  logger,
-		Address: fmt.Sprintf(":%s", addr),
-		TypeSenseParams: &search.TypeSenseParams{
-			Port:   help.GetEnv("TYPESENSE_PORT", "8108"),
-			Host:   help.GetEnv("TYPESENSE_HOST", "localhost"),
-			ApiKey: help.GetEnv("TYPESENSE_APIKEY", "xyz"),
-		},
-		StorageProvider: storageProvider,
+	var config *metadata.Config
+	if disableTypesense == "false" {
```
Comment (Sami1309) [inline metadata/server/server.go:32]: config = &metadata.Config{
			Logger:          logger,
			Address:         fmt.Sprintf(":%s", addr),
			StorageProvider: storageProvider,
		}
repeats in either block. Should it be declared before and then TypeSennseParams can just be added in a single if block after to the config?

File: charts/featureform/charts/feature-server/templates/deployment.yaml [hunk 3]
```diff
@@ -29,7 +29,7 @@ spec:
             - containerPort: {{ .Values.metrics.port }}
           env:
             - name: SERVING_PORT
-              value: 0.0.0.0:{{ .Values.serving.port }}
+              value: "{{ .Values.serving.port }}"
```
Comment (Sami1309) [inline charts/featureform/charts/feature-server/templates/deployment.yaml:32]: why make this a different convention than metadata port?

**Review by Sami1309 — COMMENTED** [review]
lgtm, just minor comments

File: charts/featureform/charts/feature-server/templates/deployment.yaml (hunk 3 above)
Comment (sdreyer) [reply charts/featureform/charts/feature-server/templates/deployment.yaml]: changed to be consistent

File: metadata/server/server.go (hunk 2 above)
Comment (sdreyer) [reply metadata/server/server.go]: removed duplicate code

File: newserving/Dockerfile (hunk 1 above)
Comment (sdreyer) [reply newserving/Dockerfile]: fixed

**Pull request merged**